protocol/ntpshm/ntpshm.go

```diff
@@ -62,14 +63,22 @@ func Create() (uintptr, error) {
 	return shmID, nil
 }
 
-func ptrToNTPSHM(shmptr uintptr) *NTPSHM {
-	var shm []NTPSHM
-	sh := (*reflect.SliceHeader)(unsafe.Pointer(&shm))
-	sh.Data = shmptr
-	sh.Len = 1
-	sh.Cap = 1
+func ptrToBytes(shmptr uintptr) []byte {
+	// Slice memory layout
+	var sl = struct {
+		addr uintptr
+		len  int
+		cap  int
+	}{shmptr, NTPSHMSize, NTPSHMSize}
 
-	return &shm[0]
+	return *(*[]byte)(unsafe.Pointer(&sl))
+}
+
+func bytesToNTPSHM(b []byte) (*NTPSHM, error) {
```

Comment:
maybe name it `ptrToNTPSHM` and call `ptrToBytes` from here to avoid calling both functions afterwards? No strong opinion anyway